dvc/external_repo.py

```diff
@@ -62,16 +62,33 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     # Adjust new clone/copy to fit rev and cache_dir
     repo = Repo(new_path)
+    original_repo = None
     try:
         if rev is not None:
             repo.scm.checkout(rev)
+        # check if the URL is local and no default remote
+        # add default remote pointing to the original repo's cache location
+        if not os.path.isdir(url):
+            # Adjust original repo for pointing remote towards its' cache
+            original_repo = Repo(url)
+            rconfig = RemoteConfig(original_repo.config)
+            if not _remote_config_exists(rconfig):
+                rconfig.add(
+                    "upstream",
+                    original_repo.cache.local.cache_dir,
+                    default=True,
```

Comment:
Better set it at the local level, same as we do with `cache_dir` down below.
```suggestion
                    default=True,
                    level=Config.LEVEL_LOCAL,
```